cdap-api/src/main/java/co/cask/cdap/api/workflow/WorkflowToken.java

```diff
@@ -23,39 +25,80 @@
 /**
  * Interface to represent the data that is transferred from one node to the next nodes in the {@link Workflow}.
  */
+@Beta
 public interface WorkflowToken {
 
   /**
-   * Put the specified key-value entry into the {@link WorkflowToken}.
+   * Keys in the {@link WorkflowToken} can be added by user, using the
+   * {@link WorkflowToken#put} method. These keys are added under the {@link Scope#USER} scope.
+   * CDAP also adds some keys to the {@link WorkflowToken}. for e.g. MapReduce counters.
+   * The keys added by CDAP gets added under {@link Scope#SYSTEM} scope.
+   */
+  public enum Scope {
+    USER,
+    SYSTEM
+  }
+
+  /**
+   * Put the specified key and value into the {@link WorkflowToken}.
    * The token may store additional information about the context in which
    * this key is being set, for example, the unique name of the workflow node.
-   * @param key   the key representing the entry
+   * @param key the key representing the entry
    * @param value the value for the key
    */
+  // TODO [CDAP-2895] put operation should throw certain exceptions
```

Comment:
Can remove this TODO from here, no? We can fix that jira later, but I don't think it applies here, since its going to be a runtime exception?